flit_core/flit_core/common.py

```diff
@@ -28,18 +28,22 @@ def __init__(self, name, directory='.'):
         if osp.isdir(pkg_dir):
             self.path = pkg_dir
             self.is_package = True
+            self.prefix = ''
             existing.add(pkg_dir)
         elif osp.isfile(py_file):
```

Comment:
I just noticed that these `elif` should be `if` - we're checking that only one of the possible paths exists, but `elif` assumes that's the case and bypasses the check.

It's not related to this change, but if you're feeling keen, another PR to add a failing test and fix that would be welcome. If not, I'll get round to it at some point.

Reply:
Does this mean that we want to implement additional checks to make sure the choices are mutually exclusive?